flowbased-computation/flowbased-computation-impl/src/main/java/com/farao_community/farao/flowbased_computation/impl/FlowbasedComputationImpl.java

```diff
@@ -288,6 +300,35 @@ private DataMonitoredBranch buildDataMonitoredBranch(FlowCnec cnec, ZonalData<Li
                 ).collect(Collectors.toList());
     }
 
+    /**
+     * Apply all remedial actions saved in CRAC, on a given network, at a given state.
+     *
+     * @param network Network on which remedial actions should be applied
+     * @param crac CRAC that should contain result extension
+     * @param state State for which the RAs should be applied
+     */
+    public static void applyAllNetworkRemedialActionsForState(Network network, Crac crac, State state) {
+        crac.getNetworkActions().forEach(na -> {
+            UsageMethod usageMethod = na.getUsageMethod(state);
+            if (usageMethod.equals(UsageMethod.AVAILABLE) || usageMethod.equals(UsageMethod.FORCED)) {
+                na.apply(network);
+            }
+        });
+    }
+
+    /**
+     * Apply remedial actions saved in CRAC result extension on current working variant of given network, at a given state.
+     *
+     * @param network Network on which remedial actions should be applied
+     * @param raoResult Result of Rao computation
+     * @param state State for which the RAs should be applied
+     */
+    public static void applyRemedialActionsForState(Network network, RaoResult raoResult, State state) {
+        raoResult.getActivatedNetworkActionsDuringState(state).forEach(na -> na.apply(network));
```

Comment:
careful that this method only returns network actions that were activated _during_ the given state.
for example, if the state is a curative one, it will not return network actions that were activated during preventive RAO.
If you want to access all network actions that were activated _during or before_ the given state, use RaoResult.isActivated(State state, NetworkAction networkAction) instead (see javadoc for more info).

Reply:
Thanks, is the new version better?

Reply:
yes perfect